### PR TITLE
added board nrf51 dk

### PR DESCRIPTION
**Pull request by sdalu**
board nrf51dk (pca10028)

### CODE REVIEW

File: boards/nrf51dk/board.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
```
Comment (haukepetersen) [inline boards/nrf51dk/board.c]: feel free to change this to 2016 and put your name/company in as copyright holder...

**Comment by haukepetersen** [comment]
Code is looking good! Two things:
- you might want to change the copyright notices in your files to 2016 and your name/company
- could you please remove the script files in the `nrf51dk/dist/` folder and use the shared jlink script (`RIOT/dist/tools/jlink/jlink.sh`) instead?! You basically only have to change your `Makefile.include`, instead of
```
+# define flash and debugging environment
 +export FLASHER = $(RIOTBOARD)/$(BOARD)/dist/flash.sh
 +export DEBUGGER = $(RIOTBOARD)/$(BOARD)/dist/debug.sh
 +export DEBUGSERVER = JLinkGDBServer -device nrf51822 -if SWD
 +export RESET = $(RIOTBOARD)/$(BOARD)/dist/reset.sh
 +
 +export OFLAGS = -O binary
 +export HEXFILE = $(ELFFILE:.elf=.bin)
 +export TERMFLAGS += -p "$(PORT)"
 +export FFLAGS = $(BINDIR) $(HEXFILE)
 +export DEBUGGER_FLAGS = $(BINDIR) $(ELFFILE)
 +export RESET_FLAGS = $(BINDIR)
```
you can just write:
```
# setup JLink for flashing
export JLINK_DEVICE := nrf51822
export JLINK_SPEED := 1000
export JLINK_PRE_FLASH := "w4 4001e504 1"
include $(RIOTBOARD)/Makefile.include.jlink
```

You might also try to ommit the speed value and see if the default value of 2000 also works...

**Comment by OlegHahm** [comment]
@haukepetersen, are you okay?

File: boards/nrf51dk/Makefile.include [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# define the used CPU
+export CPU = nrf51
+export CPU_MODEL = nrf51x22xxac
+
+# define the default port depending on the host OS
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(shell ls -1 /dev/tty.SLAB_USBtoUART* | head -n 1)
+
+export JLINK_DEVICE := nrf51822
+#export JLINK_SPEED := 1000
```
Comment (haukepetersen) [inline boards/nrf51dk/Makefile.include]: just remove if not needed

**Comment by haukepetersen** [comment]
looks good to me. Could you please remove the un-used line for setting the JLINK_SPEED and squash to 3 commits with valid commit messages?! I would propose 1 commit for the board, the CPU changes and the Jlink fix, each...

**Comment by haukepetersen** [comment]
@sdalu are you ok with my proposed changes?

Remark this for the 2016.07 release

**Comment by sdalu** [comment]
I don't know how to do it

**Comment by BytesGalore** [comment]
@sdalu you can reset your commits using `git reset HEAD~4`, which rewinds your currently 4 commits and leaves you with all changes you made until now (so nothing is lost or deleted)
Then you can `git add` the files individually in new commit groups as proposed by @haukepetersen.

I hope this helps a bit :)

**Comment by sdalu** [comment]
Thanks for the tip. But as I have unfortunately destroyed the initial repository (from unknown repository), I guess it's too late.

**Comment by sdalu** [comment]
See https://github.com/RIOT-OS/RIOT/pull/5260

**Comment by sdalu** [comment]
Master branch changed too much since initial pull request, I will not port to it. If no one is interested you can close the pull request.

**Comment by PeterKietzmann** [comment]
@sdalu, really no chance you give it one more try to update this PR :/ ?

**Comment by PeterKietzmann** [comment]
It's a pity :-( . I won't take this over now. Therefore I'll close your PR with memo label set. @sdalu we'd be happy if you take on your work again someday!

**Pull request closed without merging**